frontend/app/src/composables/history/events/mapping.ts

```diff
@@ -0,0 +1,249 @@
+          ).toString() || toSentenceCase(data.label)
+      }));
+    }
+  );
+
+  const historyEventTypeGlobalMapping: ComputedRef<HistoryEventTypeMapping> =
+    computed(() => get(HistoryEventTypeData).globalMappings);
+
+  const historyEventTypePerProtocolMapping: ComputedRef<
+    Record<string, Record<string, HistoryEventTypeMapping>>
+  > = computed(() => get(HistoryEventTypeData).perProtocolMappings);
+
+  const getEventType = (event: {
+    eventType?: string | null;
+    eventSubtype?: string | null;
+    counterparty?: string | null;
+    location?: string | null;
+  }): ComputedRef<string | undefined> =>
+    computed(() => {
+      const { eventType, eventSubtype, counterparty, location } = event;
+
+      const eventTypeNormalized = eventType || 'none';
+      const eventSubtypeNormalized = eventSubtype || 'none';
+
+      if (location && counterparty) {
+        const subTypesFromPerProtocolMapping = get(
+          historyEventTypePerProtocolMapping
+        )[location]?.[counterparty]?.[eventTypeNormalized]?.[
+          eventSubtypeNormalized
+        ];
+
+        if (subTypesFromPerProtocolMapping) {
+          return subTypesFromPerProtocolMapping;
+        }
+      }
+
+      return (
+        get(historyEventTypeGlobalMapping)[eventTypeNormalized]?.[
+          eventSubtypeNormalized
+        ] ?? undefined
+      );
+    });
+
+  const getEventTypeData = (
+    event: {
+      eventType?: string | null;
+      eventSubtype?: string | null;
+      counterparty?: string | null;
+      location?: string | null;
+    },
+    showFallbackLabel = true
+  ): ComputedRef<ActionDataEntry> =>
+    computed(() => {
+      const type = get(getEventType(event));
+
+      if (type) {
+        return get(transactionEventTypesData).find(
+          (data: ActionDataEntry) =>
+            data.identifier.toLowerCase() === type.toLowerCase()
+        )!;
+      }
+
+      const unknownLabel = tc('backend_mappings.events.type.unknown');
+      const label = showFallbackLabel
+        ? event.eventSubtype || event.eventType || unknownLabel
+        : unknownLabel;
+
+      return {
+        identifier: '',
+        label,
+        icon: 'mdi-help',
+        color: 'red'
+      };
+    });
+
+  const getEventCounterpartyData = (
+    event: HistoryEventEntry,
+    scrambler?: (hex: string) => string
+  ): ComputedRef<ActionDataEntry | null> =>
+    computed(() => {
+      const { counterparty } = event;
+      const excludedCounterparty = ['gas'];
+
+      if (counterparty && excludedCounterparty.includes(counterparty)) {
+        return null;
+      }
+
+      if (counterparty && !isValidEthAddress(counterparty)) {
+        const data = get(historyEventCounterpartiesData).find(
+          (data: ActionDataEntry) => {
+            if (data.matcher) {
+              return data.matcher(counterparty);
+            }
+            return data.identifier.toLowerCase() === counterparty.toLowerCase();
+          }
```

Comment:
```suggestion
            ({ identifier, matcher }: ActionDataEntry) => {
              if (matcher) {
                return matcher(counterparty);
              }
              return identifier.toLowerCase() === counterparty.toLowerCase();
            }     
```

A small nitpick here (indentation is hard to get write on copy/paste), can we change it to destructure that parameter